src/session_manager.rs

```diff
@@ -764,6 +764,28 @@ impl SessionManager {
         }
     }
 
+    pub(crate) fn handle_deeplink(&self, url: String) {
+        let client_id = self.active_logged_in_client_id();
+        if let Some(client_id) = client_id {
+            let username = url
+                .replace("tg://", "")
+                .replace("resolve?domain=", "")
+                .split('&')
+                .next()
+                .unwrap()
+                .to_owned();
+
+            //TODO: Open chat at given message_id
+            spawn(clone!(@weak self as obj => async move {
+                let result = functions::search_public_chat(username, client_id).await;
+                if let Ok(chat) = result {
+                    let enums::Chat::Chat(chat) = chat;
+                    obj.select_chat(client_id, chat.id);
+                }
+            }));
```

Comment:
This should be done in a totally different way. TDLib provides helpers for us to handle links in a complete way.

First, you should call [get_internal_link_type()](https://docs.rs/tdlib/latest/tdlib/functions/fn.get_internal_link_type.html) to know the [type of the link](https://docs.rs/tdlib/latest/tdlib/enums/enum.InternalLinkType.html). Then you can add support for the right type of link. In this case you probably want to support the [PublicChat](https://docs.rs/tdlib/latest/tdlib/enums/enum.InternalLinkType.html#variant.PublicChat) type.

Also, I suggest putting all this link management code inside of `session.rs`.

Reply:
Thanks for your review, now I can see that all I was doing was a hack and I've almost forgot how much generous tdlib is.